File: app/src/main/java/de/hype/hypenotify/core/BootReceiver.java
package de.hype.hypenotify.core;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class BootReceiver extends BroadcastReceiver {

    private static final String LAST_BOOT_TIME = "last_boot_time";

    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            DynamicIntents.startBackgroundService(context);
        }
    }

}

File: app/src/main/java/de/hype/hypenotify/app/core/BootReceiver.kt
package de.hype.hypenotify.app.core

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent) {
        if (Intent.ACTION_BOOT_COMPLETED == intent.getAction()) {
            DynamicIntents.Companion.startBackgroundService(context)
        }
    }

    companion object {
        private const val LAST_BOOT_TIME = "last_boot_time"
    }
}
